qutebrowser/config/configtypes.py

```diff
@@ -1029,17 +1029,27 @@ def to_py(self, value: _StrUnset) -> typing.Union[configutils.Unset,
             openparen = value.index('(')
             kind = value[:openparen]
             vals = value[openparen+1:-1].split(',')
-            int_vals = [self._parse_value(v) for v in vals]
-            if kind == 'rgba' and len(int_vals) == 4:
-                return QColor.fromRgb(*int_vals)
-            elif kind == 'rgb' and len(int_vals) == 3:
-                return QColor.fromRgb(*int_vals)
-            elif kind == 'hsva' and len(int_vals) == 4:
-                return QColor.fromHsv(*int_vals)
-            elif kind == 'hsv' and len(int_vals) == 3:
-                return QColor.fromHsv(*int_vals)
-            else:
-                raise configexc.ValidationError(value, "must be a valid color")
+
+            converters = {
+                'rgba': QColor.fromRgb,
+                'rgb': QColor.fromRgb,
+                'hsva': QColor.fromHsv,
+                'hsv': QColor.fromHsv,
+            }
+
+            conv = converters.get(kind)
+            if not conv:
+                raise configexc.ValidationError(
+                    value,
+                    '{} not in {}'.format(kind, list(converters.keys())))
```

Comment:
dict keys are unordered in Python 3.5, so the error message in the test doesn't match. Also, note that converting a dict to a list gives you the keys by default, so I'd do:

```suggestion
                    '{} not in {}'.format(kind, list(sorted(converters))))
```

(and adjust the test to match)